### PR TITLE
Replace all __call__ methods in Links with forward

### PR DESCRIPTION
**Pull request by mitmul**
By the request from @okuta , I picked up some part of changes from #4730 

**We need to check all the documents and examples carefully.**

### CODE REVIEW

**Comment by niboshi** [comment]
This PR fixes #2748.

File: chainer/link.py [hunk 1]
```diff
@@ -142,6 +142,10 @@ def __init__(self, **params):
             shape, dtype = _ensure_shape_dtype(value)
             self.add_param(name, shape, dtype=dtype)
 
+    def __call__(self, *args, **kwargs):
+        out = self.forward(*args, **kwargs)
```
Comment (okuta) [inline chainer/link.py]: How about one liner code?

Comment (mitmul) [reply chainer/link.py]: forward_preprocess and forward_postprocess will be inserted before and after this line. So I just want to leave it as is.

Comment (niboshi) [reply chainer/link.py]: >forward_preprocess and forward_postprocess will be inserted before and after this line. So I just want to leave it as is.

That can be done in that PR.
Each PR should be self-consistent.

Comment (mitmul) [reply chainer/link.py]: Why isn't this self-consistent? I don't think it breaks "self-consistency".

Comment (niboshi) [reply chainer/link.py]: >forward_preprocess and forward_postprocess will be inserted before and after this line. So I just want to leave it as is.

Doesn't this mean it's imperfect until forward_preprocess and forward_postprocess will be inserted?

Comment (okuta) [reply chainer/link.py]: OK, I accept current code.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**